sdk/search/search-documents/src/serviceUtils.ts

```diff
@@ -268,13 +267,17 @@ export function convertFieldsToPublic(fields: GeneratedSearchField[]): SearchFie
   return fields.map<SearchField>((field) => {
     let result: SearchField;
     if (field.type === "Collection(Edm.ComplexType)" || field.type === "Edm.ComplexType") {
-      result = field as ComplexField;
+      return {
```

Comment:
would it be possible to add a quick unit test for this case to guard against regression?

Reply:
Done with the latest commit